packages/yew/src/platform/mod.rs

```diff
@@ -0,0 +1,65 @@
+//! This module provides io compatibility over browser tasks and other asyncio runtimes (e.g.:
+//! tokio).
+//!
+//! Yew implements a single-threaded runtime that executes `!Send` futures. When your application
+//! starts with `yew::Renderer` or is rendered by `yew::ServerRenderer`, it is executed within the
+//! Yew runtime. The renderer will select a worker thread from the internal worker
+//! pool of Yew runtime. All tasks spawned with `spawn_local` will run on the same worker thread as
+//! the rendering thread the renderer has selected. When the renderer runs in a WebAssembly target,
+//! all tasks will be scheduled on the main thread.
+//!
+//! Yew runtime is implemented with native runtimes depending on the target platform and can use
+//! all features (timers / IO / task synchronisation) from the selected native runtime:
+//!
+//! - `wasm-bindgen-futures` (WebAssembly targets)
+//! - `tokio` (non-WebAssembly targets)
+//!
+//! Yew runtime alleviates the implementation requirement of `Send` futures when running with
+//! multi-threaded runtimes like `tokio` and `!Send` futures on WebAssembly platforms and produces
+//! good performance when the workload is IO-bounded and have similar runtime cost. When you have an
+//! expensive CPU-bounded task, it should be spawned with a `Send`-aware spawning mechanism provided
+//! by the native runtime, `std::thread::spawn` or `yew-agent` and communicates with the application
+//! using channels or agent bridges.
+
+use std::future::Future;
+
+#[cfg(feature = "ssr")]
+pub(crate) mod sync;
+
+#[cfg(not(any(feature = "tokio", target_arch = "wasm32")))]
+#[path = "rt_null/mod.rs"]
+mod imp;
+#[cfg(all(not(target_arch = "wasm32"), feature = "tokio"))]
+#[path = "rt_tokio/mod.rs"]
+mod imp;
+#[cfg(target_arch = "wasm32")]
+#[path = "rt_wasm_bindgen/mod.rs"]
+mod imp;
```

Comment:
We need to build docs in a way that the functions provided by these modules show up. Right now, [they do not](https://yew-rs-api--pr2697-platform-qxx7puat.web.app/next/yew/platform/index.html).

Can you update the `cargo doc` command in the build rustdoc workflow accordingly?

Reply:
This is due to that the `sync` module is `pub(crate)`.

It currently only implements the methods that is actually used by Yew.
The actual channel implementation should have more methods.
I think when if open this up to other users, we should have a complete implementation.

Reply:
I think you misunderstood. If you look at the docs page, there's no `run_pinned` function or `sync` module (that's what I meant by the exposed functions). Those should be visible in docs.

Reply:
`run_pinned` is `pub(crate)`; what part of `sync` would make sense to expose for downstream users?

Reply:
It doesn't make sense anymore, now that sync is simply a wrapper around tokio's implementation. I mentioned it when run_pinned was still pub. It's all good now